### ISSUE TITLE
Polybar transparency breaks Compton composition

### ISSUE THREAD

**Issue by Kmambu** [polybar/polybar#800]
Hi there,

I recently decided to switch from a solid background (RGB) to a transparent one (ARGB) and I noticed some big issues with the way Compton handles it. Most notably, when I go back from sleep, Compton freeze after a few seconds, and before that, I can sometimes spot artifacts on the Polybar before everything goes frozen, which makes me suspect my polybar config is at fault here

Here is my polybar config : 
;=====================================================
;
;   To learn more about how to configure Polybar
;   go to https://github.com/jaagr/polybar
;
;   The README contains alot of information
;
;=====================================================

[colors]
foreground = #cccccc
foreground-alt = #000000
background = #ab000000

primary = #ffb52a
secondary = #e60053
alert = #bd2c40

[settings]
;screenchange-reload = true
;compositing-background = xor
;compositing-foreground = clear
;compositing-overline = xor
;compositing-underline = xor
;compositing-border = xor
format-foreground = ${colors.foreground}
;format-background = ${colors.background}

[bar/topbar]
width = 100%
height = 30
bottom = false
offset-x = 0%
offset-y = 0%
radius = 0
fixed-center = true

background = ${colors.background}
foreground =  ${colors.foreground}

padding-left = 1
padding-right = 1

module-margin-left = 1
module-margin-right = 1

tray-offset-x = 0%
tray-offset-y = 0%
tray-maxsize = 18
tray-transparent = true
tray-detached = false
tray-background = ${colors.background}

font-0 = Sauce Code Powerline:antialias=true:pixelsize=12
font-1 = SauceCodePro Nerd Font:antialias=true:pixelsize=18

modules-left = battery0 battery1 xkeyboard
modules-center = i3
modules-right = pavolume date wifi-network

tray-position = 
tray-padding = 15

wm-restack = i3

[module/battery0]
format-foreground = ${colors.background}
type = internal/battery
full-at = 99
battery = BAT0
adapter = AC
label-charging =%percentage%%
format-charging = %{A1:mate-power-statistics &:}<animation-charging> <label-charging>%{A}

label-discharging =%percentage%%
format-discharging = %{A1:mate-power-statistics &:}<ramp-capacity> <label-discharging>%{A}
label-full = 
ramp-capacity-0 = 
ramp-capacity-1 = 
ramp-capacity-2 = 
ramp-capacity-3 = 
ramp-capacity-4 = 

animation-charging-0 = 
animation-charging-1 = 
animation-charging-2 = 
animation-charging-3 = 
animation-charging-4 = 
[module/battery1]
inherit = module/battery0
battery = BAT1

[module/wifi-network]
type = internal/network
interface = wlp3s0
accumulate-stats = true
label-connected = " %downspeed%"
label-disconnected =  wired
format-connected = %{A1:networkmanager_dmenu &:}<label-connected>%{A}
format-disconnected = %{A1:networkmanager_dmenu &:}<label-disconnected>%{A}

[module/pavolume]
type = custom/script
tail = true
label = %output%
exec = ~/.config/polybar/pavolume.sh --listen
click-left = exec pavucontrol
click-middle = ~/.config/polybar/pavolume.sh --togmute
scroll-up = ~/.config/polybar/pavolume.sh --up
scroll-down = ~/.config/polybar/pavolume.sh --down

[module/xkeyboard]
type = internal/xkeyboard
blacklist-0 = num lock

format-prefix = " "

label-layout = %layout%
label-indicator-padding = 0
label-indicator-margin = 1

[module/i3]
ws-icon-0 = 1;
ws-icon-1 = 2;
ws-icon-2 = 3;
ws-icon-3 = 4;
ws-icon-4 = 5;
ws-icon-5 = 6;
ws-icon-6 = 7;
ws-icon-7 = 8;
ws-icon-8 = 9;
ws-icon-9 = 10;

label-focused = %icon%
label-unfocused = %icon%
label-urgent = %icon%
label-visible = %icon%

type = internal/i3
format = <label-state> <label-mode>
index-sort = true
wrapping-scroll = false

workspace-label = %icon%
; pin-workspaces = true

label-mode-padding = 0
label-focused-padding = 2
label-unfocused-padding = 2
label-urgent-padding = 2

label-focused-underline = ${colors.foreground-alt}
label-focused-foreground = ${colors.foreground-alt}
label-focused-background = ${colors.foreground}

[module/date]
type = internal/date
interval = 5

date =
date-alt = " %Y-%m-%d"

time = %H:%M
time-alt = %H:%M:%S

format-prefix = 

label = %date% %time%

[global/wm]
margin-top = 5
margin-bottom = true 5

**Comment by patrick96** [comment]
Does the freezing also happen when polybar is not running?
When you say, freeze, do you mean the whole desktop is unresponsive or that just compton stopped working?

**Comment by Kaegell** [comment]
The entire desktop is unresponsive, and much, I cannot switch TTYs (or at least, it is not shown since the monitor view is frozen as well). I tried to get a log from compton but did not manage to.

I have this issue only since I use the ARGB format on the background of my polybar.

In the meanwhile, I switched compton for a fork supposed to fix the GLX backend issues, and everything is working fine for now. But I don't know why a GLX Backend compositor would break on polybar :/

I'll open a ticket at the repository of compton as well
